client.go

```diff
@@ -266,13 +266,18 @@ func (r remoteProcess) Close() error {
 
 func joinErrs(errs ...error) error {
 	var str string
+	foundErr := false
 	for _, e := range errs {
 		if e != nil {
+			foundErr = true
 			if str != "" {
 				str += ", "
 			}
 			str += e.Error()
 		}
 	}
-	return xerrors.New(str)
+	if foundErr {
+		return xerrors.New(str)
+	}
+	return nil
```

Comment:
Good catch. Is it easier though to just do this before the loop?
I just dislike `found` bools in loops. Either way, good fix

```golang
if len(errs) == 0 {
  return nil
}

```

Reply:
We can't simplify to that because we need `joinErrs(nil, nil, nil)` to return `nil`